### ISSUE TITLE
REST framework field doesn't work with model validation

### ISSUE THREAD

**Issue by jonashaag** [django-money/django-money#376]
Scenario: Want to add MinValueValidator to model money field.

A Decimal based minimum value doesn't work because on saving a model instance, djmoney complains it can't compare the decimal minimum value with the instance's Money object (understandably).

A Money based minimum value doesn't work because then REST framework field expects Decimal validators, so it complains it can't compare the Money minimum value with the REST request's Decimal value.

The only way to fix this is to repeat the Money field definition in the REST framework serializer class and make it use a second version of the MinValueValidator that is Decimal based.

I think we should be using Money based validation in the REST framework field instead of Decimal based. And make the REST Money field return Money objects rather than Decimal all the time (not only if currency is explicitly given in request).

**Comment by danielpellon** [comment]
I also experience the same issue, could you share an example of your workaround?

**Comment by webyneter** [comment]
@jonashaag can you share the workaround?

**Comment by jonashaag** [comment]
Don't have the code anymore sorry but it should be pretty easy to implement.

**Comment by webyneter** [comment]
For anyone else who stumbles upon the same stone, here's my workaround:

```python
from functools import partial

from django.conf import settings
from djmoney.contrib.django_rest_framework import MoneyField as drf_MoneyField
from djmoney.models.fields import MoneyField as model_MoneyField
from djmoney.models.validators import MinMoneyValidator

_money_field_kwargs = dict(
    max_digits=settings.CURRENCY_MAX_DIGITS,
    decimal_places=settings.CURRENCY_DECIMAL_PLACES,
    validators=[MinMoneyValidator(0)],
)
ProjectMoneyField = partial(model_MoneyField, **_money_field_kwargs)
# region Refrain from using validators for the DRF field until
# https://github.com/django-money/django-money/issues/376
# is fixed.
ProjectAPIMoneyField = partial(drf_MoneyField, **{**_money_field_kwargs, "validators": []})
# endregion
```

Note that `settings.CURRENCY_MAX_DIGITS` and `settings.CURRENCY_DECIMAL_PLACES` are set in my Django project settings.

**Comment by benjaoming** [comment]
If someone can contribute a fix to this (preferably with a test), we'll easily get it merged and released :1st_place_medal:

**Comment by Stranger6667** [comment]
Should be fixed via #467
